src/vs/code/electron-main/menubar.ts

```diff
@@ -495,22 +474,57 @@ export class Menubar {
 		}
 	}
 
-	private setMenuById(menu: Electron.Menu, menuId: string): void {
-		console.log(`Attempting to set menu for ${menuId}`);
-
-		// Build dynamic menu
-		this.menubarMenus[menuId].items.forEach((item: IMenubarMenuItemAction | IMenubarMenuItemSeparator) => {
-			if (item.id === 'vscode.menubar.separator') {
+	private setMenu(menu: Electron.Menu, items: Array<MenubarMenuItem>) {
+		items.forEach((item: MenubarMenuItem) => {
+			if (isMenubarMenuItemSeparator(item)) {
 				menu.append(__separator__());
-			} else {
-				let menuItem: Electron.MenuItem;
-				let action: IMenubarMenuItemAction = <IMenubarMenuItemAction>item;
-				menuItem = this.createMenuItem(action.label, action.id, action.enabled, action.checked);
+			} else if (isMenubarMenuItemSubmenu(item)) {
+				const submenu = new Menu();
+				const submenuItem = new MenuItem({ label: this.mnemonicLabel(item.label), submenu: submenu });
+				this.setMenu(submenu, item.submenu.items);
+				menu.append(submenuItem);
+			} else if (isMenubarMenuItemAction(item)) {
+				if (item.id === 'workbench.action.openRecent') {
```

Comment:
I would leave a TODO here since it feels a bit hacky how we are inserting recent menu items

Reply:
Yea, we do this on both sides for the custom and native menubar experience. maybe we should discuss with a joh a way to do this better with the MenuRegistry in the future

Reply:
Would you like me to create a debt item so we do not forget about this
fyi @jrieken

Reply:
what's that about? dynamically computed menu entries?